stable/stolon/templates/secret.yaml

```diff
@@ -8,18 +7,17 @@ metadata:
     chart: {{ template "stolon.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+  annotations:
+    "helm.sh/resource-policy": keep
```

Comment:
Can you explain why you remove `{{ if .Release.IsInstall }}` and add the annotation instead?

Reply:
@unguiculus  I'm still trying to sort this out.  Intent is to create the secret on initial install and maintain the secrete data with subsequent `helm upgrade`s.   With `{{ if .Release.IsInstall }}` the secret is deleted during `helm upgrade`.  I thought the removal of `{{ if .Release.IsInstall }}` + addition of annotation would remedy, but it does not appear so.

see [gist](https://gist.github.com/rtluckie/7ee6dac40ce6cca91c78fbbb4196bde1) for details.

Reply:
The annotation doesn't help here because you don't delete the release.

I would remove `{{ if .Release.IsInstall }}` and live with the fact that auto-generated passwords are regenerated on upgrade. Many charts have this problem. Using `--reuse-values` could help here. Alternatively, remove default passwords completely and make them `required`. You can provide a custom values file with passwords for CI only.

See https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values.